### PR TITLE
Refactoring/Optimizing Installer/Install usage

### PR DESCRIPTION
**Pull request by nupplaphil**
Redux & Prerequisite of #6815 

I refactored the `Installer` core class and the `Install` module, so we use the config Cache for states and no more check on $_POST variables inside content functions :-)

In #6815, I use this base for the "real" enhancements :-)

- [x] console.php currently not working (see #6879 why)

### CODE REVIEW

File: src/Module/Install.php [hunk 1]
```diff
@@ -61,45 +62,51 @@ public static function init()
 	public static function post()
 	{
 		$a = self::getApp();
+		$configCache = $a->getConfigCache();
 
 		switch (self::$currentWizardStep) {
 			case self::SYSTEM_CHECK:
+				self::checkSetting($configCache, $_POST, 'config', 'php_path');
+				break;
```
Comment (MrPetovan) [inline src/Module/Install.php]: Why did you add these two lines? They're the same as the next case, you can leave them combined like
```php
case self::SYSTEM_CHECK:
case self::DATABASE_CONFIG:
    self::checkSetting($configCache, $_POST, 'config', 'php_path');
    break;
```

**Comment by nupplaphil** [comment]
- Tested local with wizard (and with false db credentials)
- Tested with automatic installer
- Enhanced tests

File: src/Core/Console/AutomaticInstallation.php [hunk 2]
```diff
@@ -207,8 +220,8 @@ private function runBasicChecks(Installer $installer)
 		}
 
 		$php_path = null;
-		if (!empty(Config::get('config', 'php_path'))) {
-			$php_path = Config::get('config', 'php_path');
+		if (!$configCache->has('config', 'php_path')) {
```
Comment (nupplaphil) [inline src/Core/Console/AutomaticInstallation.php]: wrong boolean!

**Comment by nupplaphil** [comment]
Have to fix the mentioned bug first (But I'm currently at work)
